lsp/gopls.lua

```diff
@@ -23,23 +90,9 @@ return {
   filetypes = { 'go', 'gomod', 'gowork', 'gotmpl' },
   root_dir = function(bufnr, on_dir)
     local fname = vim.api.nvim_buf_get_name(bufnr)
+    get_mod_cache_dir()
+    get_std_lib_dir()
     -- see: https://github.com/neovim/nvim-lspconfig/issues/804
-    if mod_cache then
-      on_dir(get_root(fname))
-      return
-    end
-    local cmd = { 'go', 'env', 'GOMODCACHE' }
-    vim.system(cmd, { text = true }, function(output)
-      if output.code == 0 then
-        if output.stdout then
-          mod_cache = vim.trim(output.stdout)
-        end
-        on_dir(get_root(fname))
-      else
-        vim.schedule(function()
-          vim.notify(('[gopls] cmd failed with code %d: %s\n%s'):format(output.code, cmd, output.stderr))
-        end)
-      end
-    end)
+    on_dir(get_root_dir(fname))
```

Comment:
FYI if a dir is NOT found, it's generally best to NOT call `on_dir`. Otherwise the config will be activated even if `workspace_required=true`.